src/Servers/HttpSys/src/FeatureContext.cs

```diff
@@ -619,10 +619,10 @@ private async Task NotifiyOnStartingAsync()
                 return;
             }
 
-            actions.Reverse();
             // Execute last to first. This mimics a stack unwind.
-            foreach (var actionPair in actions)
+            for (var i = actions.Count; i-- > 0; )
```

Comment:
```suggestion
            for (var i = actions.Count; i > 0; i--)
```
Is the "typical" way to write loops in C#.

But I don't care (just that it's said 😉).

Reply:
"typical" is a very subjective term.

That's a typical for loop for me.

Up until variables can be introduced in expressions. Then, it will be something like this:

```C#
while (var i = actions.Count; i-- > 0)
{
    // ..
}
```

Reply:
I'm pretty sure that every for-loop within dotnet-org (and beyond) is written in the way as given in the [language reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/for). So we should stick to that convention.

Personally I find it harder to read in the form `for (var i = n; i-- > 0; )` as year long training for the eyes match `for (var i = n; i > 0; --i)` quite better.

Besides all that, the for-loop as written has bug (index violation) which is the most important thing to watch for.

Reply:
Check the examples at https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/for#examples

This use case is not there, but it doesn't make it less valid.

There's no `-->` in C#.

The expression is: `(i--) > 0`

Reply:
> There's no `-->` in C#.

I know 😉 In some other languages the motivation to write loops like this, comes from the look-a-like of the variable going down to 0, so `--> 0` (analogy to limit).

But again, this is unusual in C# and especially in the dotnet-org and we should stay consistent.

For this point I've said enough and won't respond for this.

> Where is the bug?

In my brain...see other comment.

Reply:
Again, it's a style preference and "unusual" is in the eye of the beholder.

I'm OK with accepting a style that is equivalent in function and performance. I won't accept that there's anything wrong or unusual with the style I used.

Reply:
In this case "beholder" is a whole team responsible to keep code not necessarily the best looking ever but consistent. There is nothing wrong with the you wrote your loop It's a valid code with no bugs. But it's just inconsistent with other code in this repo. And keeping thing consistent in a large code base is what helps people get through different pieces of code without trying to understand it's code conventions anew every time

Reply:
That's fair, and I already accepted that more than once.

Let's wait for someone that can to make the final decision.

Reply:
> Let's wait for someone that can to make the final decision.

As maintainer of this repo I am asking you to please change it

Reply:
Sure! Can I have a reason for that?

Reply:
To be consistent with other for loops in the repo.

Reply:
Done! Sorry for the delay.